src/pages/admin/ProductListPage/index.tsx

```diff
@@ -0,0 +1,119 @@
+import { useEffect, useState } from 'react';
+import { FormProvider, useForm } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import { NavLink } from 'react-router-dom';
+
+import { Box, CircularProgress } from '@mui/material';
+
+import { sortOptions, urls } from 'src/common/constants';
+import useErrorHandling from 'src/common/hooks/useErrorHandlingHook';
+import SearchInput from 'src/components/shared/SearchInput';
+import useToast from 'src/components/shared/toasts/components/ToastProvider/ToastProviderHooks';
+import { useGetAllProductsQuery } from 'src/redux/adminProduct/adminProductService';
+import { SortOrder } from 'src/redux/user/types';
+import theme from 'src/theme';
+
+import AdminSectionSubTitle from '../AdminSectionSubTitle';
+import AdminSectionTitle from '../AdminSectionTitle';
+import {
+  SectionWrapper,
+  StyledListItemButton,
+  StyledTypography,
+} from '../ProductRequestPage/styles';
+import ProductsTable from '../ProductsTable';
+import SortButton from '../SortButton';
+
+function ProductListPage() {
+  const { t } = useTranslation();
+  const { showToast } = useToast();
+  const methods = useForm();
+
+  const [page, setPage] = useState(1);
+  const handleChange = (_: React.ChangeEvent<unknown>, value: number) => {
+    setPage(value);
+  };
+
+  const [sortOrder, setSortOrder] = useState<SortOrder>(sortOptions.DESC);
+  const handleClick = (value: SortOrder) => {
+    setSortOrder(value);
+  };
+
+  const [search, setSearch] = useState('');
+  const handleSearchChange = (value: string) => {
+    setSearch(value);
+    setPage(1);
+  };
+
+  const { data, isLoading, error } = useGetAllProductsQuery({
+    list: 'published',
+    page,
+    sortOrder,
+    search,
+  });
+  const limit = 5;
+  const products = data?.products || [];
+  const count = data?.count || 1;
+  const pagesCount = Math.ceil(count / limit);
+
+  const { handleOnSubmitError } = useErrorHandling();
+
+  useEffect(() => {
+    if (error) {
+      handleOnSubmitError(
+        error,
+        showToast,
+        t('productsAdmin.productListError')
+      );
+    }
+  }, [error, handleOnSubmitError, showToast, t]);
+
+  if (isLoading) {
+    return <CircularProgress />;
+  }
```

Comment:
better create a wrapper and use it for each page

Reply:
made a Loader component